lib/pages/chat/events/message_download_content.dart

```diff
@@ -1,45 +1,187 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:dartz/dartz.dart' hide State, OpenFile;
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/utils/manager/download_file_state.dart';
+import 'package:fluffychat/utils/manager/download_manager.dart';
+import 'package:fluffychat/utils/matrix_sdk_extensions/download_file_extension.dart';
 import 'package:fluffychat/utils/matrix_sdk_extensions/event_extension.dart';
+import 'package:fluffychat/utils/storage_directory_utils.dart';
+import 'package:fluffychat/widgets/file_widget/download_file_tile_widget.dart';
 import 'package:fluffychat/widgets/file_widget/file_tile_widget.dart';
 import 'package:fluffychat/widgets/file_widget/message_file_tile_style.dart';
+import 'package:fluffychat/widgets/mixins/handle_download_and_preview_file_mixin.dart';
 import 'package:flutter/material.dart';
 import 'package:matrix/matrix.dart';
 
-class MessageDownloadContent extends StatelessWidget {
+class MessageDownloadContent extends StatefulWidget {
   final Event event;
-  final void Function(Event event)? onFileTapped;
 
   final String? highlightText;
 
   const MessageDownloadContent(
     this.event, {
     Key? key,
-    this.onFileTapped,
     this.highlightText,
   }) : super(key: key);
 
   @override
-  Widget build(BuildContext context) {
-    final filename = event.filename;
-    final filetype = event.fileType;
-    final sizeString = event.sizeString;
+  State<MessageDownloadContent> createState() => _MessageDownloadContentState();
+}
+
+class _MessageDownloadContentState extends State<MessageDownloadContent>
+    with HandleDownloadAndPreviewFileMixin {
+  final downloadManager = getIt.get<DownloadManager>();
+
+  final downloadProgressNotifier = ValueNotifier<double?>(0);
+
+  final downloadFileStateNotifier =
+      ValueNotifier(DownloadFileState.notDownload);
+
+  StreamSubscription<Either<Failure, Success>>? streamSubscription;
+
+  ValueNotifier<String> filePathNotifier = ValueNotifier('');
 
-    Logs().i(
-      'filename: $filename, filetype: $filetype, sizeString: $sizeString, content: ${event.content}',
+  String? downloadPath;
+
+  @override
+  void initState() {
+    super.initState();
+    streamSubscription = downloadManager
+        .eventIdMapDownloadFileInfo[widget.event.eventId]?.streamSubscription;
+    streamSubscription?.resume();
+    streamSubscription?.onData((event) {
+      onDownloadingProcess(event);
+    });
+    loadFilePath();
+  }
+
+  void loadFilePath() async {
+    downloadPath =
+        await StorageDirectoryUtils.instance.getDownloadFolderInApp();
+    final filePath = '$downloadPath/${widget.event.filename}';
+    if (await File(filePath).exists() && streamSubscription == null) {
```

Comment:
create another stream for checking file existed, then merge the result with downloading subscribtion to update view. (use only one ValueNotifier)